### PR TITLE
[test] Job log collection as a single action

### PR DESCRIPTION
**Pull request by sebsoto**
Collects job logs in one go, rather than using two API calls, one to gather the logs to print to a file, and one to return the logs to the caller.

### CODE REVIEW

File: test/e2e/network_test.go
Comment (saifshaikh48) [inline test/e2e/network_test.go:138]: `%v` if you log `err` directly. alternatively `err.Error()` with `%s`

applies throughout

File: test/e2e/network_test.go
Comment (mansikulkarni96) [inline test/e2e/network_test.go:880]: No arg specified for %s

File: test/e2e/network_test.go
Comment (sebsoto) [reply test/e2e/network_test.go:138]: What is the functional difference given the error interface implements `String()`?

Comment (saifshaikh48) [reply test/e2e/network_test.go]: for some reason i thought this would result in a type error when called, tested it out and found no issue

**Review by mansikulkarni96 — APPROVED** [review]

File: test/e2e/network_test.go
Comment (saifshaikh48) [inline test/e2e/network_test.go:847]: should this be set outside the `if` as it was before? otherwise, in the case we hit our retryCount, the `gatherPodLogs()` outside the loop will be fed an empty string for a labelSelector

**Review by saifshaikh48 — COMMENTED** [review]
/hold

**Comment by sebsoto** [comment]
Considering this is an e2e PR to address flakes, overriding to unblock other PRs

**Comment by jrvaldes** [comment]
@sebsoto  consider overriding the failed jobs as they passed in the past

https://prow.ci.openshift.org/pr-history/?org=openshift&repo=windows-machine-config-operator&pr=2293

**Pull request merged**
